### ISSUE TITLE
Crash in php with UTF-8 character

### ISSUE THREAD

**Issue by albert-github** [doxygen/doxygen#6917]
When having a php script with in it something like (i.e. used for lowercase conversion in Symfony_Standard_Vendors_3.4.24; see test case at the end):
```
<?php

return array(
  'à' => 'a',
);
```
where one of the characters between the single quotes is a multi-byte UTF-8 character doxygen crashes on windows (2013 and 2017 compiler, local master build and local 1.8.15 build) with:

![crash_2017](https://user-images.githubusercontent.com/5223533/55719213-001d5b80-59fe-11e9-91fc-882ef582d71f.jpg)

It looks like this is due to the `isspace` in : 
```
      case '@':  // '@name' -> ' @name'
      case '$':  // '$name' -> ' $name'
      case '\'': // ''name' -> '' name'
        if (i>0 && i<l-1 && pc!='=' && pc!=':' && !isspace(pc) &&
            isId(nc) && osp<8) // ")id" -> ") id"
        {
          *dst++=' ';
        }
        *dst++=c;
        break;
```
in routine `removeRedundantWhiteSpace` in `util.cpp`

The case: [symphony_crash.zip](https://github.com/doxygen/doxygen/files/3053955/symphony_crash.zip)

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #6922

**Comment by albert-github** [comment]
Code has been integrated in master on github (please don't close the issue as this will be done at the moment of a release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.16.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
